src/lib/ConfirmationController.js

```diff
@@ -59,10 +59,26 @@ export default class ConfirmationController {
     return document.querySelector(this.#config.dialogSelector)
   }
 
+  get acceptButtons() {
+    this.dialogTarget.querySelectorAll(this.#config.acceptSelector)
+  }
+
+  get denyButtons() {
+    this.dialogTarget.querySelectorAll(this.#config.denySelector)
+  }
+
   #showConfirm() {
     // if this is the first time, store the HTML of the dialog.
     // We'll use this to restore the dialog to its original state on teardown.
-    if (!this.#initialContent) this.#initialContent = this.dialogTarget.innerHTML
+    if (!this.#initialContent) {
+      this.#initialContent = this.dialogTarget.innerHTML
+
+      // add a listener for the cancel event on the dialog, to fully deny the confirmation
+      this.dialogTarget.addEventListener('cancel', (event) => {
+        event.preventDefault();
+        this.denyButtons.forEach(element => element.click())
+      });
```

Comment:
I'm not that familiar with the dialog element, but I'd prefer this live in `#setupListeners` if possible.  Any reason why it couldn't?  I'll comment there with a suggestion.